src/Uno.UI/UI/Xaml/Controls/TextBlock/TextVisual.skia.cs

```diff
@@ -154,7 +155,16 @@ internal override void Render(SKSurface surface, SKImageInfo info)
 
 				foreach (var line in _textLines)
 				{
-					surface.Canvas.DrawText(line, 0, y, _paint);
+					// Workaround DrawShapedText crashing sometimes when the _paint.GetFont().typefont (internal to SkiaSharp) is null.
```

Comment:
Do you know in which condition this can happen? This feels like we should know more about the conditions when this can happen.

This change replaces the whole text rendering stack, and for the better. It's looking pretty good. Do you know if this could be a replacement solution for https://github.com/unoplatform/uno/pull/6975 ?

Reply:
@jeromelaban I don't think any of them can be a replacement for the other. Both PRs solve a different problem. The other issue is that the font doesn't support the character.

Reply:
> Do you know in which condition this can happen? This feels like we should know more about the conditions when this can happen.

Not much that I know, but the crash happens in SkiaSharp at this line:

https://github.com/mono/SkiaSharp/blob/2ad29861d5a40d3bf78c28ab0a9cb02a8f0fe437/source/SkiaSharp.HarfBuzz/SkiaSharp.HarfBuzz.Shared/CanvasExtensions.cs#L10-L17

when `paint.GetFont().Typeface` returns null.

Reply:
We'll need to open an issue there then, I don't feel comfortable integrating something where there's an arbitrary try/catch.

Reply:
@jeromelaban I think there could be some way to detect that beforehand. Will it be okay if we use an if condition instead of a try/catch?